x-pack/plugins/ingest_manager/public/applications/ingest_manager/components/settings_flyout.tsx

```diff
@@ -131,8 +131,8 @@ export const SettingFlyout: React.FunctionComponent<Props> = ({ onClose }) => {
 
   useEffect(() => {
     if (settings) {
-      inputs.kibanaUrl.setValue(
-        settings.kibana_url || [`${window.location.origin}${core.http.basePath.get()}`]
+      inputs.kibanaUrls.setValue(
+        settings.kibana_urls || [`${window.location.origin}${core.http.basePath.get()}`]
```

Comment:
I wonder if this should check if `settings.kibana_urls` length > 0? (not sure if it is possible to have empty array here)

Reply:
It's a little bit confusing to me, @nchaulet might be able to answer better, but i'm not sure why the attributes on the settings saved object type are optional.  The urls are [set during setup](https://github.com/elastic/kibana/blob/master/x-pack/plugins/ingest_manager/server/services/setup.ts#L79) and we only allow them to update attributes not overwrite them through the PUT /settings api.

Reply:
I think we can change the `kibana_urls` typing to be required at's is now done in the setup. (we iterated a few time on this and settings were optional at some point)